### PR TITLE
Remove GP5 from automation tests

### PR DESCRIPTION
**Pull request by djianwen-vmware**
- Because PXF7 will deprecate GP5 support, we no longer need special case handling for GP5 in the automation test
- Removed the replacements from "DETAIL: " to "CONTEXT: " in the expected answer files and SQL files. Additional explanations:
  - GP6 uses "CONTEXT:" instead of "DETAIL:" in the output for more information
  - So in the past we replace almost all "DETAIL:" with "CONTEXT:" such that we can verify GP5 and GP6 with the same expected output file
  - Now we don't need them any more. GP6 and GP7 are consistent in this scenario.
- Removed version checks of GP5 in Java automation test code
- No longer suppress the "HINT: " for GP5. Additional explanations:
  - GP5 doesn't support hint, so we disabled the check in the past.
  - Now "HINT:" output will be validated during automation test
- Added missing `HINT:` in some of automation test cases
- Removed GP5 automation test cases
- Removed unnecessary “start_ignore”, “end_ignore”, “start_matchsubs”, and “end_matchsubs” in automation tests as a result of the removals above
- Fixed orc write `primitive_types_array_multi` automation test which failed because of nondeterministic results
  - In the past we suppress the "HINT:" so we didn't realize the hint in the error message is nondeterministic in this test case. It depends on the distribution of the table and thus each time the hint will be different and causes test failures.
  - added `id=` such that the result is deterministic.

### CODE REVIEW

**Review by denalex — APPROVED** [review]

**Review by tatb-vmware — APPROVED** [review]

File: automation/sqlrepo/features/hive/errors/columnDataTypeMisMatch/expected/query01.ans [hunk 1]
```diff
@@ -7,10 +7,7 @@
...
 --
--- m/DETAIL/
--- s/DETAIL/CONTEXT/
---
 -- end_matchsubs
 SELECT * from pxf_hive_small_data ORDER BY t1;
 ERROR:  For field t1 schema requires type INTEGER but input record has type TEXT
-DETAIL:  External table pxf_hive_small_data
\ No newline at end of file
+CONTEXT:  External table pxf_hive_small_data
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/hive/errors/columnDataTypeMisMatch/expected/query01.ans:13]: can we add new line at end of this file

File: automation/sqlrepo/features/hive/errors/columnDataTypeMisMatch/sql/query01.sql [hunk 2]
```diff
@@ -7,8 +7,5 @@
 -- m/(ERROR|WARNING):.*  External table definition did not match input data: column.*seg.*/
 -- s/seg.*//
 --
--- m/DETAIL/
--- s/DETAIL/CONTEXT/
---
 -- end_matchsubs
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/hive/errors/columnDataTypeMisMatch/sql/query01.sql:10]: can we add new line at the end of this file

File: automation/sqlrepo/features/hive/errors/columnNameMismatch/expected/query01.ans [hunk 3]
```diff
@@ -7,13 +7,8 @@
...
 --
--- m/DETAIL/
--- s/DETAIL/CONTEXT/
---
 -- end_matchsubs
 SELECT * from pxf_hive_small_data ORDER BY t1;
 ERROR:  PXF server error : column 's2' does not exist in the Hive schema
--- start_ignore
 HINT:  Ensure the column exists and check the column name spelling and case. Check the PXF logs located in the 'logs-dir' directory on host 'mdw' or 'set client_min_messages=LOG' for additional details.
--- end_ignore
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/hive/errors/columnNameMismatch/expected/query01.ans:18]: new line at end of file

File: automation/sqlrepo/features/hive/errors/columnNameMismatch/sql/query01.sql [hunk 4]
```diff
@@ -7,8 +7,5 @@
 -- m/   description   .*/
 -- s/description   .*/description   DESCRIPTION/
 --
--- m/DETAIL/
--- s/DETAIL/CONTEXT/
---
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/hive/errors/columnNameMismatch/sql/query01.sql:12]: new line at end of file

File: automation/sqlrepo/features/hive/errors/partitionNameMismatch/expected/query01.ans [hunk 5]
```diff
@@ -7,16 +7,11 @@
...
---
 -- m/CONTEXT:.*line.*/
 -- s/line \d* of //g
 --
 -- end_matchsubs
 SELECT * from pxf_hive_small_data ORDER BY t1;
 ERROR:  PXF server error : column 's2' does not exist in the Hive schema or Hive Partition
--- start_ignore
 HINT:  Ensure the column or partition exists and check the name spelling and case. Check the PXF logs located in the 'logs-dir' directory on host 'mdw' or 'set client_min_messages=LOG' for additional details.
--- end_ignore
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/hive/errors/partitionNameMismatch/expected/query01.ans:21]: new line at end of file

File: automation/sqlrepo/features/jdbc/readable_date_wide_range/expected/query01.ans [hunk 6]
```diff
@@ -9,9 +7,6 @@
 -- m/(.*)ERROR:/
 -- s/(.*)ERROR:/ERROR:/
 --
--- m/DETAIL/
--- s/DETAIL/CONTEXT/
---
 -- m/CONTEXT:.*line.*/
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/jdbc/readable_date_wide_range/expected/query01.ans:10]: can we remove this match sub as well

File: automation/sqlrepo/features/orc/write/primitive_types_array_multi/sql/query01.sql [hunk 7]
```diff
@@ -20,11 +17,11 @@
...
-INSERT INTO orc_primitive_arrays_multi_bool_writable SELECT id, bool_arr FROM orc_primitive_arrays_multi_heap;
-INSERT INTO orc_primitive_arrays_multi_bytea_writable SELECT id, bytea_arr FROM orc_primitive_arrays_multi_heap;
-INSERT INTO orc_primitive_arrays_multi_int_writable SELECT id, int_arr FROM orc_primitive_arrays_multi_heap;
-INSERT INTO orc_primitive_arrays_multi_float_writable SELECT id, float_arr FROM orc_primitive_arrays_multi_heap;
-INSERT INTO orc_primitive_arrays_multi_date_writable SELECT id, date_arr FROM orc_primitive_arrays_multi_heap;
+INSERT INTO orc_primitive_arrays_multi_bool_writable SELECT id, bool_arr FROM orc_primitive_arrays_multi_heap WHERE id = 1;
+INSERT INTO orc_primitive_arrays_multi_bytea_writable SELECT id, bytea_arr FROM orc_primitive_arrays_multi_heap WHERE id = 0;
+INSERT INTO orc_primitive_arrays_multi_int_writable SELECT id, int_arr FROM orc_primitive_arrays_multi_heap WHERE id = 0;
+INSERT INTO orc_primitive_arrays_multi_float_writable SELECT id, float_arr FROM orc_primitive_arrays_multi_heap WHERE id = 0;
+INSERT INTO orc_primitive_arrays_multi_date_writable SELECT id, date_arr FROM orc_primitive_arrays_multi_heap WHERE id = 0;
```
Comment (axue-broadcom) [inline automation/sqlrepo/features/orc/write/primitive_types_array_multi/sql/query01.sql:24]: new line at end of file

File: automation/sqlrepo/proxy/hive_small_data/expected/query04.ans [hunk 8]
```diff
@@ -38,10 +33,8 @@ CREATE ROLE
 \connect - testuser
 SELECT * FROM pxf_proxy_hive_small_data_allowed_no_impersonation ORDER BY name;
 ERROR:  PXF server error : Permission denied: user=foobar, access=READ, inode=SOME_PATH:-rwx------
--- start_ignore
 HINT:  Check the PXF logs located in the 'logs-dir' directory on host 'mdw' or 'set client_min_messages=LOG' for additional details.
--- end_ignore
-DETAIL:  External table pxf_proxy_hive_small_data_allowed_no_impersonation
+CONTEXT:  External table pxf_proxy_hive_small_data_allowed_no_impersonation
```
Comment (axue-broadcom) [inline automation/sqlrepo/proxy/hive_small_data/expected/query04.ans:37]: new line at end of file

File: automation/sqlrepo/proxy/hive_small_data/expected/query05.ans [hunk 9]
```diff
@@ -38,10 +33,8 @@ CREATE ROLE
 \connect - testuser
 SELECT * FROM pxf_proxy_hive_small_data_prohibited_no_impersonation ORDER BY name;
 ERROR:  PXF server error : Permission denied: user=foobar, access=READ, inode=SOME_PATH:-rwx------
--- start_ignore
 HINT:  Check the PXF logs located in the 'logs-dir' directory on host 'mdw' or 'set client_min_messages=LOG' for additional details.
--- end_ignore
-DETAIL:  External table pxf_proxy_hive_small_data_prohibited_no_impersonation
+CONTEXT:  External table pxf_proxy_hive_small_data_prohibited_no_impersonation
```
Comment (axue-broadcom) [inline automation/sqlrepo/proxy/hive_small_data/expected/query05.ans:37]: new line at end of file

File: automation/sqlrepo/proxy/hive_small_data/sql/query04.sql [hunk 10]
```diff
@@ -14,9 +14,6 @@
 -- m/pxf:\/\/pxf_automation_data\/proxy\/([0-9a-zA-Z]).*\/data.txt/
 -- s/pxf:\/\/pxf_automation_data\/proxy\/([0-9a-zA-Z]).*\/data.txt/pxf:\/\/pxf_automation_data\/proxy\/OTHER_USER\/data.txt/
 --
--- m/DETAIL/
--- s/DETAIL/CONTEXT/
---
```
Comment (axue-broadcom) [inline automation/sqlrepo/proxy/hive_small_data/sql/query04.sql:19]: similar question here, can we remove the bottom 2 lines

File: automation/sqlrepo/proxy/hive_small_data_ipa/expected/query04.ans [hunk 11]
```diff
@@ -38,10 +33,8 @@ CREATE ROLE
 \connect - testuser
 SELECT * FROM pxf_proxy_hive_ipa_small_data_allowed_no_impersonation ORDER BY name;
 ERROR:  PXF server error : Permission denied: user=foobar, access=READ, inode=SOME_PATH:-rwx------
--- start_ignore
 HINT:  Check the PXF logs located in the 'logs-dir' directory on host 'mdw' or 'set client_min_messages=LOG' for additional details.
--- end_ignore
-DETAIL:  External table pxf_proxy_hive_ipa_small_data_allowed_no_impersonation
```
Comment (axue-broadcom) [inline automation/sqlrepo/proxy/hive_small_data_ipa/expected/query04.ans:44]: new line at end of file

File: automation/sqlrepo/proxy/hive_small_data_ipa/expected/query05.ans [hunk 12]
```diff
@@ -38,10 +33,8 @@ CREATE ROLE
 \connect - testuser
 SELECT * FROM pxf_proxy_hive_ipa_small_data_prohibited_no_impersonation ORDER BY name;
 ERROR:  PXF server error : Permission denied: user=foobar, access=READ, inode=SOME_PATH:-rwx------
--- start_ignore
 HINT:  Check the PXF logs located in the 'logs-dir' directory on host 'mdw' or 'set client_min_messages=LOG' for additional details.
--- end_ignore
-DETAIL:  External table pxf_proxy_hive_ipa_small_data_prohibited_no_impersonation
```
Comment (axue-broadcom) [inline automation/sqlrepo/proxy/hive_small_data_ipa/expected/query05.ans:44]: new line at end of file

**Review by axue-broadcom — APPROVED** [review]

**Pull request merged**